rfcs/0000-since-parameter.md

```diff
@@ -0,0 +1,141 @@
+
+### BFD Feed Details
+
+The ETL server will write to a S3 bucket a feed of NDJSON records. Each record contains fields for the ETL job's id, status, start and end timestamps. The job id is an integer. The timestamps follows the FHIR specification for instant formating. The record also contains a pointer to a list of beneficiaries which were udpated in the ETL job. The list is a UTF-8 encoded comma seperated list of MBIs. The MBI list will be written in a sorted in order. 
+
+```
+{
+    "id": 100,
+    "status": "done",
+    "start": "2019-02-07T12:28:17-05:00",
+    "end": "2019-02-07T13:28:17-05:00" 
+    "beneficiaries": "s3://bfd-prod-etl-feed/bene20191101.mbi"
+}
+```
+The S3 bucket with the feed and the MBI lists are private, with only access to BFD partners. S3 cross-account access policies will be used to lock down access. 
+
+### Since Implementors Details
+
+To add since support, downstream BFD partners must:
+1. Update their export operation end-point to support '_since.' 
+2. At the start of a job, read the ETL feed from S3. From this feed retrieve: 
+   1. The start and end time of the last ETL job
+   2. The list of beneficiaries updated in the last ETL job. 
+3. Based on the timestamp passed in the `_since` parameter, optimize the queries to BFD in the following ways: 
+   1. If BFD did not run ETL jobs since the passed timestamp, do not call the BFD and return an empty result set. 
+   2. If BFD ran one ETL job since the passed-in timestamp, call BFD only for those beneficiaries included in the beneficiary list.
+   3. If BFD ran multiple jobs since the passed-in timestamp, it is acceptable not to optimize BFD calls and call the BFD for all beneficiaries in the export job.  
+
+The scalability of BFD could be a limiting factor in the expansion of DPC and BCDA to more customers. Every week less than 5% of beneficiaries have a new claim. By only calling BFD when a beneficiary has an updated claim, the partner is effectively increasing the capacity of BFD.
+
+### ETL Corner Case
+
+Every export job completion record contains a `transactionTime` field that bulk-export clients use as the `_since` parameter in a subsequent export job. The `transactionTime` is the time that the export job starts. The FHIR bulk export specification states that an export SHALL only contain resources updated before the `transactionTime` time. The specification further says that an exporter should delay a job until all pending writes have finished to satisfy this constraint. The ELT feed allows bulk-export implementors to know when the BFD ETL finishes. Since both the bulk export operation and the BFD ETL process take several hours, the delay may be significant. 
+
+To avoid this delay, a bulk-export implementor may take an optimistic approach by immediately starting an export job, but monitoring if it receives a resource updated after the job start time. If it does, the it should delay and restart the job.  
```

Comment:
Would requesting resources where `lastUpdated` is between the time specified in the `_since` parameter and the export `transactionTime`?

Reply:
Consider the case where the same record was updated in consecutive loads: after the since time passed in and again after the bulk export job was started (ie. the transactionTime). In this case, the record where would have a lastUpdate time after the transactionTime, so it would not be included in the download, when it should have been.